### PR TITLE
Updated documentation according to new defaults for idiomatic endpoints

### PR DESCRIPTION
**Pull request by L-Lavigne**
This PR adapts the documentation to the changes in https://github.com/higherkindness/skeuomorph/pull/342 and https://github.com/higherkindness/sbt-mu-srcgen/pull/103. Specifically, it changes the description of `muSrcGenIdiomaticEndpoints` in `microsite/src/main/docs/reference/source-generation.md`  and adds documentation for `namespace`  and `methodNameStyle` in `microsite/src/main/docs/tutorials/grpc-server-client.md`.

It also removes some redundant default values in `@service` annotations, tweaks the language hints in the fenced code blocks, fixes broken tables, breaks overly long lines, and removes newlines inside link labels.

### CODE REVIEW

File: microsite/src/main/docs/guides/custom-grpc-serialization.md [hunk 1]
```diff
@@ -28,21 +28,18 @@ To make the server compress its responses, set the compression type argument to
 
 For example:
 
-```scala mdoc:silent
+```scala
```
Comment (juanpedromoreno) [inline microsite/src/main/docs/guides/custom-grpc-serialization.md:31]: *question*: is this intentional? I guess we still want to run docs at CI, right?

Comment (L-Lavigne) [reply microsite/src/main/docs/guides/custom-grpc-serialization.md]: @juanpedromoreno It wasn't really intentional; I had actually _added_ a bunch of `mdoc` tags but several caused failures and I kept getting new errors as I reverted them one at a time. Then I lost patience and just removed them all, including a few previously-existing ones by accident. I just pushed a PR to restore those, hopefully this works.

File: microsite/src/main/docs/tutorials/testing-rpc-service.md [hunk 2]
```diff
@@ -158,7 +156,7 @@ class ServiceSpec extends AnyFlatSpec with ServiceAndClient {
 
 Let's see the test in action:
 
-```scala mdoc
+```scala
```
Comment (juanpedromoreno) [inline microsite/src/main/docs/tutorials/testing-rpc-service.md:159]:
```suggestion
```scala mdoc
```

File: microsite/src/main/docs/tutorials/testing-rpc-service.md [hunk 3]
```diff
@@ -197,6 +195,6 @@ class PropertyBasedServiceSpec extends AnyFlatSpec with ServiceAndClient with Ch
 
 Let's run this test as well:
 
-```scala mdoc
+```scala
```
Comment (juanpedromoreno) [inline microsite/src/main/docs/tutorials/testing-rpc-service.md:198]:
```suggestion
```scala mdoc
```

**Review by juanpedromoreno — APPROVED** [review]

**Pull request merged**